### PR TITLE
Make description.md self contained.

### PR DESCRIPTION
**Pull request by Insti**
Resolves: #767

The problem description given in description.md should be complete and self contained, and not require the addition of the blurb to make sense.

This patch prepends the blurb from the `metadata.yml` onto the `description.md` together with a newline, matching the format provided by trackler for the README.

Waiting on: Update to trackler so that readmes so that the blurb does not appear to appear twice.

### CODE REVIEW

**Comment by rpottsoh** [comment]
@Insti not sure what you mean by `waiting on other tracks`.  I am not clear on what the other tracks need to do.  For my own understanding, this link (for example) `http://exercism.io/languages/<track>/exercises` will end up showing the first lines from the description.md files instead of the blurb from the metadata.yml files?  It is also my understanding that a blank line is required between the "blurb" and the rest of the description in the description.md file.  Going forward will Travis CI be checking new description.md files to make sure there is a blank line?

**Comment by Insti** [comment]
`waiting on other tracks` applies to trackler. https://github.com/exercism/trackler/issues/42, but not other language tracks, does this mean that it is an inappropriate label?

The blank line between the current blurb and the rest of the description is not required, but it is to retain compatibility with the currently displayed readme. Going forward I expect descriptions to be edited and can envision this blank line disappearing.

>  For my own understanding, this link (for example) http://exercism.io/languages/<track>/exercises will end up showing the first lines from the description.md files instead of the blurb from the metadata.yml files?

No, the description shown on that page will still come from the blurb in `metadata.yml`

**Comment by Insti** [comment]
The script has been run on all the existing `description.md` files.

The necessary changes to trackler have been merged: https://github.com/exercism/trackler/pull/43

This PR can now be merged. 
The changes will be included in the following trackler release.

cc: @kytrinyx

File: exercises/counter/description.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+Design a test suite for a line/letter/character counter tool.
```
Comment (stkent) [inline exercises/counter/description.md]: Does is matter that this line was copied in above the deprecation notice?

Comment (Insti) [inline exercises/counter/description.md]: Good spot. 

That is the location required so that the README will still be rendering identically.

Comment (stkent) [inline exercises/counter/description.md]: Cool, thanks for clarifying for me :)

**Comment by Insti** [comment]
I'd like to get a few approvals before I pull the trigger on merging this.
I think it's ready to go but I'm worried I've missed something.

File: exercises/acronym/description.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+Convert a long phrase to its acronym
```
Comment (ErikSchierboom) [inline exercises/acronym/description.md]: Does it have to be a long phrase? And perhaps add a dot at the end of the line.

File: exercises/bank-account/description.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+Simulate a bank account supporting opening/closing, withdraws, and deposits of money. Watch out for concurrent transactions!
```
Comment (ErikSchierboom) [inline exercises/bank-account/description.md]: Should withdraws be withdrawals?

File: exercises/binary/description.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+Convert a binary number, represented as a string (e.g. '101010'), to its decimal equivalent using first principles
```
Comment (ErikSchierboom) [inline exercises/binary/description.md]: Missing dot at the end of the line.

File: exercises/book-store/description.md [hunk 5]
```diff
@@ -1,3 +1,5 @@
+To try and encourage more sales of different books from a popular 5 book series, a bookshop has decided to offer discounts of multiple-book purchases.
+
```
Comment (ErikSchierboom) [inline exercises/book-store/description.md]: It seems these lines are duplicated below, albeit with a slightly different wording.

File: exercises/bowling/description.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+Score a bowling game
```
Comment (ErikSchierboom) [inline exercises/bowling/description.md]: Missing dot at the end of the line.

File: exercises/bracket-push/description.md [hunk 7]
```diff
@@ -1,2 +1,4 @@
+Make sure the brackets and braces all match.
+
```
Comment (ErikSchierboom) [inline exercises/bracket-push/description.md]: Duplicate lines below.

File: exercises/change/description.md [hunk 8]
```diff
@@ -1,3 +1,5 @@
+Correctly determine change to be given using the least number of coins
+
```
Comment (ErikSchierboom) [inline exercises/change/description.md]: Missing dot at the end of the line. Also duplicate description below, although the description below does give more information.

File: exercises/circular-buffer/description.md [hunk 9]
```diff
@@ -1,3 +1,5 @@
+A data structure that uses a single, fixed-size buffer as if it were connected end-to-end.
+
```
Comment (ErikSchierboom) [inline exercises/circular-buffer/description.md]: Duplicate description.

File: exercises/clock/description.md [hunk 10]
```diff
@@ -1,3 +1,5 @@
+Implement a clock that handles times without dates.
+
```
Comment (ErikSchierboom) [inline exercises/clock/description.md]: The line below is not necessary I think.

File: exercises/connect/description.md [hunk 11]
```diff
@@ -1,3 +1,5 @@
+Compute the result for a game of Hex / Polygon
```
Comment (ErikSchierboom) [inline exercises/connect/description.md]: Missing dot at the end of the line.

File: exercises/dot-dsl/description.md [hunk 12]
```diff
@@ -1,3 +1,5 @@
+Write a Domain Specific Language similar to the Graphviz dot language
```
Comment (ErikSchierboom) [inline exercises/dot-dsl/description.md]: Missing dot at the end of the line.

File: exercises/error-handling/description.md [hunk 13]
```diff
@@ -1,3 +1,5 @@
+Implement various kinds of error handling and resource management
```
Comment (ErikSchierboom) [inline exercises/error-handling/description.md]: Missing dot at the end of the line.

File: exercises/flatten-array/description.md [hunk 14]
```diff
@@ -1,3 +1,5 @@
+Take a nested list and return a single list with all values except nil/null
```
Comment (ErikSchierboom) [inline exercises/flatten-array/description.md]: Missing dot at the end of the line. Maybe add the word flatten after single?

Comment (ErikSchierboom) [inline exercises/flatten-array/description.md]: I mean flattened.

File: exercises/food-chain/description.md [hunk 15]
```diff
@@ -1,3 +1,5 @@
+Generate the lyrics of the song 'I Know an Old Lady Who Swallowed a Fly'
```
Comment (ErikSchierboom) [inline exercises/food-chain/description.md]: Missing dot at the end of the line.

File: exercises/forth/description.md [hunk 16]
```diff
@@ -1,3 +1,5 @@
+Implement an evaluator for a very simple subset of Forth
```
Comment (ErikSchierboom) [inline exercises/forth/description.md]: Missing dot at the end of the line.

File: exercises/grade-school/description.md [hunk 17]
```diff
@@ -1,3 +1,5 @@
+Given students' names along with the grade that they are in, create a roster for the school
```
Comment (ErikSchierboom) [inline exercises/grade-school/description.md]: Missing dot at the end of the line.

File: exercises/linked-list/description.md [hunk 18]
```diff
@@ -1,3 +1,5 @@
+Implement a doubly linked list
```
Comment (ErikSchierboom) [inline exercises/linked-list/description.md]: Missing dot at the end of the line.

File: exercises/list-ops/description.md [hunk 19]
```diff
@@ -1,3 +1,5 @@
+Implement basic list operations
```
Comment (ErikSchierboom) [inline exercises/list-ops/description.md]: Missing dot at the end of the line.

File: exercises/minesweeper/description.md [hunk 20]
```diff
@@ -1,3 +1,5 @@
+Add the numbers to a minesweeper board
```
Comment (ErikSchierboom) [inline exercises/minesweeper/description.md]: Missing dot at the end of the line.

File: exercises/pig-latin/description.md [hunk 21]
```diff
@@ -1,3 +1,5 @@
+Implement a program that translates from English to Pig Latin
```
Comment (ErikSchierboom) [inline exercises/pig-latin/description.md]: Missing dot at the end of the line.

File: exercises/poker/description.md [hunk 22]
```diff
@@ -1,3 +1,5 @@
+Pick the best hand(s) from a list of poker hands.
+
```
Comment (ErikSchierboom) [inline exercises/poker/description.md]: Duplicate description.

File: exercises/pov/description.md [hunk 23]
```diff
@@ -1,3 +1,5 @@
+Reparent a graph on a selected node
```
Comment (ErikSchierboom) [inline exercises/pov/description.md]: Missing dot at the end of the line.

File: exercises/simple-cipher/description.md [hunk 24]
```diff
@@ -1,3 +1,5 @@
+Implement a simple shift cipher like Caesar and a more secure substitution cipher
```
Comment (ErikSchierboom) [inline exercises/simple-cipher/description.md]: Missing dot at the end of the line.

File: exercises/simple-linked-list/description.md [hunk 25]
```diff
@@ -1,3 +1,5 @@
+Write a simple linked list implementation that uses Elements and a List
```
Comment (ErikSchierboom) [inline exercises/simple-linked-list/description.md]: Missing dot at the end of the line.

File: exercises/transpose/description.md [hunk 26]
```diff
@@ -1,3 +1,5 @@
+Take input text and output it transposed.
+
```
Comment (ErikSchierboom) [inline exercises/transpose/description.md]: Duplicate description

File: exercises/protein-translation/description.md [hunk 27]
```diff
@@ -1,3 +1,5 @@
+Translate RNA sequences into proteins.
+
```
Comment (ErikSchierboom) [inline exercises/protein-translation/description.md]: Duplicate description

**Comment by Insti** [comment]
Thanks for the thorough read through @ErikSchierboom 

The current PR is an interim one that only prepends the blurb to the description.
The blurbs are where the duplication and missing punctuation come from.

I think I can fix the missing periods, but my plan is to wait until after 
https://github.com/exercism/trackler/issues/42 is completed to fix the redundancy/duplications or else it will be more difficult to synchronise the updated descriptions with trackler and we risk that every readme will end up with a duplicated first line.

**Comment by ErikSchierboom** [comment]
@Insti I kinda figured that, but this was an ideal moment to review all comments in one go. My comments can then be used later to remove duplication. The periods can also wait as far as I am concerned. Thanks for putting this together!

**Comment by Insti** [comment]
It turns out adding the periods was easy enough to do.

File: exercises/scale-generator/description.md [hunk 28]
```diff
@@ -1,4 +1,4 @@
-Generate musical scales, given a starting note and a set of intervals. 
+Generate musical scales, given a starting note and a set of intervals. .
```
Comment (Insti) [inline exercises/scale-generator/description.md]: Extra period here.

**Comment by Insti** [comment]
I think I will move the period fixes to a different PR.

File: exercises/acronym/description.md (hunk 2 above)
Comment (kytrinyx) [inline exercises/acronym/description.md]: 👍 for striking ~long~

File: exercises/bank-account/description.md (hunk 3 above)
Comment (kytrinyx) [inline exercises/bank-account/description.md]: Agreed, I think this is a typo.

**Pull request merged**